client/transaction-pool/src/lib.rs

```diff
@@ -57,16 +74,44 @@ impl<PoolApi, Block> BasicPool<PoolApi, Block>
 		PoolApi: sc_transaction_graph::ChainApi<Block=Block, Hash=Block::Hash>,
 {
 	/// Create new basic transaction pool with provided api.
-	pub fn new(options: sc_transaction_graph::Options, pool_api: PoolApi) -> Self {
+	pub fn new(
+		options: sc_transaction_graph::Options,
+		pool_api: PoolApi,
+	) -> Self {
+		Self::with_revalidation_type(options, pool_api, RevalidationType::Full)
+	}
+
+	/// Create new basic transaction pool with provided api and custom
+	/// revalidation type.
+	pub fn with_revalidation_type(
+		options: sc_transaction_graph::Options,
+		pool_api: PoolApi,
+		revalidation_type: RevalidationType,
+	) -> Self {
+		let api = Arc::new(pool_api);
+		let cloned_api = api.clone();
 		BasicPool {
-			pool: Arc::new(sc_transaction_graph::Pool::new(options, pool_api)),
+			api: cloned_api,
+			pool: Arc::new(sc_transaction_graph::Pool::new(options, api)),
+			revalidation_strategy: Arc::new(Mutex::new(
```

Comment:
Maybe better move the mutability internally to the type? Some checks can be easily done without acquiring the lock in such case.

Reply:
I redid in such way that it is no longer possible :)
But it is almost instant lock, not much more than just reading atomic